### PR TITLE
WID-79: HPC launch

### PR DESCRIPTION
**Pull request by 915-Misan-Teodora**
We implement here the **launch on HPC**.
Still not completed: 
- [ ] the serialization of a Simulator at submit time (will be another task as described in task WID-208)
- [ ] bug on remote launch exec, the result file is missing !!

### CODE REVIEW

File: tvbwidgets/core/hpc/hpc_launch.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
+import logging
```
Comment (liadomide) [inline tvbwidgets/core/hpc/hpc_launch.py:1]: Missing header in this file

File: tvbwidgets/core/hpc/hpc_launch.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    env_dir = 'tvb_widgets_t'
+    env_name = 'venv_t'
+    python_dir = {'DAINT-CSCS': 'python3.9', 'JUSUF': 'python3.10'}
+    modules = {'DAINT-CSCS': 'cray-python', 'JUSUF': 'Python'}
+    pip_libraries = 'tvb-widgets tvb-data'
+    EXECUTABLE_KEY = 'Executable'
+    PROJECT_KEY = 'Project'
+    JOB_TYPE_KEY = 'Job type'
+    INTERACTIVE_KEY = 'interactive'
+    project = 'icei-hbp-2021-0007'
```
Comment (liadomide) [inline tvbwidgets/core/hpc/hpc_launch.py:24]: some of these should go into a HPCConfig object (WID-205)

File: tvbwidgets/core/hpc/hpc_launch.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+    def __init__(self, site, param1, param2, param1_values, param2_values, metrics, file_name):
+        self.site = site
+        self.param1 = param1
+        self.param2 = param2
+        self.param1_values = param1_values
+        self.param2_values = param2_values
+        self.metrics = metrics
+        self.file_name = file_name
+        self.submit_job("parameters.py", ["C:\\Users\\teodora.misan\\Documents\\tvb-widgets\\tvbwidgets\\core\\pse\\parameters.py"], True)
```
Comment (liadomide) [inline tvbwidgets/core/hpc/hpc_launch.py:34]: Why this hard-coded input file ?

File: tvbwidgets/core/pse/parameters.py [hunk 4]
```diff
@@ -10,7 +10,8 @@
 
 .. moduleauthor:: Fousek Jan <jan.fousek@univ-amu.fr>
```
Comment (liadomide) [inline tvbwidgets/core/pse/parameters.py:11]: I think here also Teo should be explicitly mentioned as a second author

File: tvbwidgets/core/pse/parameters.py [hunk 5]
```diff
@@ -312,21 +319,21 @@ def compute_metrics(sim, metrics):
...
         else:
-            resulted_metric = (VarianceNodeVariance(sim.monitors[0].period))
+            resulted_metric = VarianceNodeVariance(sim.monitors[0].period)
         computed_metrics.append(resulted_metric)
 
     return computed_metrics
 
 
-def launch_local_param(simulator, param1, param2, x_values, y_values, metrics, file_name):
+def launch_local_param(param1, param2, x_values, y_values, metrics, file_name):
```
Comment (liadomide) [inline tvbwidgets/core/pse/parameters.py:336]: I think we can put this parameter "simulator" back, and in the __main__ of this module, temporarily give it (for the HPC run) until we have that serialised properly

**Review by liadomide — COMMENTED** [review]
As code it looks ok to me now, but I haven't manage to make it run with my account neither on JUFUS nor on JUDAC.

**Review by liadomide — APPROVED** [review]

**Comment by liadomide** [comment]
> As code it looks ok to me now, but I haven't manage to make it run with my account neither on JUFUS nor on JUDAC.

We can no share the same env-name and env-dir between users of the same project. If I changed those, it works.

**Review by rarescodemart — APPROVED** [review]

**Pull request merged**